### DISCUSSION TITLE
AODV instead of DSDV

### DISCUSSION THREAD

**Discussion by tejas-connect** [LoRaMesher/LoRaMesher#41] [Ideas]
How can we implement AODV instead of DSDV here? Any ideas

**Comment by Jaimi5** [comment]
Hi!
A new type of packet can be created to ask on demand a routing discovery to all the network.
I would add it to the TODO list.
Any good ideas?

**Comment by quintesse** [comment]
Just read up on AODV vs DSDV and I was wondering if the Managed Flood Routing that Meshtastic uses would be an option? Or do you want to support directed device to device communication with LoRaMesher? (I'm not sure what the intended use-cases are that you want LoRaMesher to support)

**Reply by Jaimi5** [reply]
Managed flood routing could be an option, but personally, I’m not a big fan of it. It typically requires forwarding a larger number of packets, which increases energy consumption—something that's especially important to consider with LoRa.

For this current version of LoRaMesher, I don’t plan to support flood-based routing. However, I’m open to experimenting with it in a future version to see how well it performs in practice.

**Reply by quintesse** [reply]
Understood. Yesterday I was reading up on [Reticulum](https://reticulum.network/), another mesh network but hardware-agnostic, where they actually mention Meshtastic and the problem with flood routing that it tends not to scale once you have a large(ish) number of nodes.

Edit: one of the reasons it seemed attractive to me is its simplicity. In my personal case it's very unlikely that I will ever get more than a handful of sensor nodes (which might be battery-operated and off most of the time) with a couple of repeater/gateways. So in that case flood routing doesn't seem that bad.

**Reply by Jaimi5** [reply]
Yes! You are right!
If you need help I could try to guide you on how I would implement flooding using LoRaMesher. It seems not very complicated to implement since, as you say, it is quite a simple mechanism.
If you want, you can join our discord and discuss furthermore there!

**Reply by quintesse** [reply]
Great, I joined the Discord 👍
